source/ada/lsp-ada_documents.adb

```diff
@@ -677,15 +721,199 @@ package body LSP.Ada_Documents is
          when Ada_For_Loop_Var_Decl |
               Ada_Label_Decl |
               Ada_Named_Stmt_Decl =>
-            return LSP.Messages.A_Constant;
+            return (if Ignore_Local
+                    then LSP.Messages.A_Null
+                    else LSP.Messages.A_Constant);
 
          when others
             => null;
       end case;
 
-      return LSP.Messages.A_Function;
+      return LSP.Messages.A_Null;
    end Get_Decl_Kind;
 
+   -----------------
+   -- Get_Profile --
+   -----------------
+
+   function Get_Profile
+     (Node : Libadalang.Analysis.Basic_Decl) return LSP.Types.LSP_String
+   is
+      use Ada.Strings.Wide_Wide_Unbounded;
+      use Libadalang.Analysis;
+      use Libadalang.Common;
+
+      function To_Text (Node : Ada_Node'Class) return Wide_Wide_String;
+      --  Retrieve the node text and format it
+
+      function To_Profile
+        (Node : Libadalang.Analysis.Subp_Spec'Class)
+         return LSP.Types.LSP_String;
+
+      -------------
+      -- To_Text --
+      -------------
+
+      function To_Text (Node : Ada_Node'Class) return Wide_Wide_String
+      is
+         Node_Text : constant String :=
+           Langkit_Support.Text.To_UTF8 (Node.Text);
+         Result    : String  := Node_Text;
+         Was_Space : Boolean := False;
+         Cur       : Integer := Node_Text'First;
+      begin
+         for I in Node_Text'Range loop
+            if Node_Text (I) = ' ' then
+               --  Trim multiple whitespace to only keep one
+               if not Was_Space then
+                  Result (Cur) := Node_Text (I);
+                  Cur := Cur + 1;
+               end if;
+               Was_Space := True;
```

Comment:
Use Ada.Strings.Fixed.Trim

Reply:
Ada.Strings.Fixed.Trim only handles whitespaces at the beginning and at the end. This is not enough for complex type given in profiles.

In the example below, Trim will not be able to handle properly the type of Bar.
   procedure Foo
       (Bar : not null access function
           (I : Integer;
           J : Float)
           return Boolean);

Reply:
Using GNATCOLL.Utils.Split would require to parse each string 2 times:
- First remove the ASCII.LF
- Second Split via whitespace (' ')
(And then we need to concatenate the list from Split)
So regarding performance it would be slower (Note: we can retrieve hundreds of profile)

Reply:
Ok, fine by me then